elixir/apps/web/lib/web/live/sign_up.ex

```diff
@@ -32,17 +33,50 @@ defmodule Web.SignUp do
         with: fn _account, attrs -> Actors.Actor.Changeset.create(attrs) end
       )
     end
+
+    defp validate_email_allowed(changeset, sign_up_enabled, allowed_domains) do
+      Ecto.Changeset.validate_change(changeset, :email, fn :email, email ->
+        cond do
+          sign_up_enabled ->
+            []
+
+          email_allowed?(email, allowed_domains) ->
+            []
+
+          true ->
+            [email: "email domain is not allowed at this time"]
+        end
+      end)
+    end
+
+    defp email_allowed?(email, allowed_domains) do
+      with [_, domain] <- String.split(email, "@", parts: 2) do
+        Enum.member?(allowed_domains, domain)
+      else
+        _ -> false
+      end
+    end
   end
 
   def mount(_params, _session, socket) do
     user_agent = Phoenix.LiveView.get_connect_info(socket, :user_agent)
     real_ip = Web.Auth.real_ip(socket)
 
     changeset =
-      Registration.changeset(%{
-        account: %{slug: "placeholder"},
-        actor: %{type: :account_admin_user}
-      })
+      Registration.changeset(
+        %{
+          account: %{slug: "placeholder"},
+          actor: %{type: :account_admin_user}
+        },
+        true,
+        []
+      )
+
+    allowed_domains =
+      Domain.Config.get_env(:domain, :sign_up_allow_list)
+      |> String.trim()
+      |> String.split(",", trim: true)
+      |> Enum.map(&String.trim/1)
```

Comment:
Just double-checking, are all of these `trim`s necessary? I.e. could you just do

```
|> String.split(",")
|> Enum.map(&String.trim/1)
```

Reply:
Good question.  I think I was so back and forth on my implementation for this whole thing yesterday that I left this in by mistake.  I think what you've proposed is good.  I might add one more pipe to reject any empty strings, like so:

```
|> String.split(",")
|> Enum.map(&String.trim/1)
|> Enum.reject(& &1 == "")
```

This is what I was ultimately trying to do, make sure that no empty strings ended up in the list and that none of the entries in the list had leading/trailing whitespace.

Reply:
@bmanifold we should do all the validations and sanitizing in the `defintions.ex` so when app boots the application environment should be clean to make sure we don't start unless we are sure we won't crash later.

Reply:
@AndrewDryga, makes sense.  Where in the definition should that be placed?  I see that the options for `defconfg` are: 
```
default: term,
sensitive: boolean(),
dump: dump_callback(),
legacy_keys: [legacy_key()],
changeset: changeset_callback()
```
but none of those seemed correct, although I could just be misunderstanding the options.